raster/r.univar/stats.c

```diff
@@ -244,17 +292,57 @@ int print_stats(univar_stat *stats)
                 }
             }
 
-            if (param.shell_style->answer) {
-                fprintf(stdout, "first_quartile=%g\n", quartile_25);
-                fprintf(stdout, "median=%g\n", median);
-                fprintf(stdout, "third_quartile=%g\n", quartile_75);
+            if (param.shell_style->answer || format == JSON) {
+                switch (format) {
+                case PLAIN:
+                    fprintf(stdout, "first_quartile=%g\n", quartile_25);
+                    fprintf(stdout, "median=%g\n", median);
+                    fprintf(stdout, "third_quartile=%g\n", quartile_75);
+                    break;
+                case JSON:
+                    json_object_set_number(zone_object, "first_quartile",
+                                           quartile_25);
+                    json_object_set_number(zone_object, "median", median);
+                    json_object_set_number(zone_object, "third_quartile",
+                                           quartile_75);
+                    break;
+                }
+
+                JSON_Value *percentiles_array_value, *percentile_value;
+                JSON_Array *percentiles_array;
+                JSON_Object *percentile_object;
+
+                if (format == JSON) {
+                    percentiles_array_value = json_value_init_array();
+                    percentiles_array = json_array(percentiles_array_value);
+                }
+
                 for (i = 0; i < stats[z].n_perc; i++) {
-                    char buf[24];
+                    char buf[24], buf2[36];
 
                     sprintf(buf, "%.15g", stats[z].perc[i]);
```

Comment:
Even if not introduced by this PR, you might change `sprintf` to the more safer `snprintf`:
```suggestion
                    snprintf(buf, sizeof(buf), "%.15g", stats[z].perc[i]);
```